lib/libcam/tests/libcam_test.c

```diff
@@ -0,0 +1,170 @@
+#include <sys/cdefs.h>
+//RCSID goes here
+
+#include <atf-c.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <camlib.h>
+
+/* cam_get_device tests */
+
+ATF_TC(libcam_get_device_null_path);
+
+ATF_TC_HEAD(libcam_get_device_null_path, tc)
+{
+	atf_tc_set_md_var(tc, "descr", "test cam_get_device with a null path string.");
+}
+
+ATF_TC_BODY(libcam_get_device_null_path, tc)
+{
+	char* path = NULL;
+	char  dev_name[10];
+	int   devnamelen = 10;
+	int   unit = 0;
+	int   result = 0;
+	struct cam_device * dev;
+
+	atf_tc_expect_fail("Passing in a null device path.");
+
+	ATF_CHECK(cam_get_device(path, dev_name, devnamelen, &unit) == 0);
+}
+
+ATF_TC(libcam_get_device_empty_path);
+
+ATF_TC_HEAD(libcam_get_device_empty_path, tc)
+{
+	atf_tc_set_md_var(tc, "descr", 
+			  "test cam_get_device with an empty path string.");
+}
+
+ATF_TC_BODY(libcam_get_device_empty_path, tc)
+{
+	char* path = "";
+	char  dev_name[10];
+	int   devnamelen = 10;
+	int   unit = 0;
+	int   result = 0;
+	struct cam_device * dev;
+
+	atf_tc_expect_fail("Passing in an empty device path.");
+
+	ATF_CHECK(cam_get_device(path, dev_name, devnamelen, &unit) == 0);
+}
+
+ATF_TC(libcam_get_device_no_text_after_slash);
+
+ATF_TC_HEAD(libcam_get_device_no_text_after_slash, tc)
+{
+	atf_tc_set_md_var(tc, "descr", 
+			  "test cam_get_device with an invalid device path that is only /.");
```

Comment:
Indentation is a bit off. It should be something like:
<pre>
\tatf_tc_set_mc_var(tc, "descr",
\t    "cam_get_device with an invalid device path: '/'");
</pre>